### PR TITLE
Fixes Everyone Is A Traitor not affecting latejoins.

### LINKED ISSUE #38463: Everyone is the traitor doesn't affect latejoins
Issue reported from  Round ID: 89410 (/tg/Station Bagil [ENGLISH] [US-EAST] [100% FREE LAG])

Reporting client version: 512

Probably fixable by doing something similar to whatever summon guns/magic does

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
I have made the admin sheanigean use a datum that registers the `COMSIG_GLOB_CREWMEMBER_JOINED` signal to give latejoins the traitor antag datum. I have also moved the global comsig at the end of `AttemptLateSpawn()`, which is necessary to stop latejoins who already rolled antag from getting the role and also make it work for non humans (though robots can't be latejoined as right now). For summon guns/magic, a `ishuman` check has been added, and the definition of the `make_scottish()` has been moved to `living`. This shouldn't affect how the other datums who use the same comsig work.

## Why It's Good For The Game
This will fix #38463.

## Changelog
:cl:
admin: Fixed the Everyone Is A Traitor admin button not affecting latejoins.
/:cl:

### CODE REVIEW

**Comment by necromanceranne** [comment]
a06

File: code/modules/antagonists/separatist/separatist.dm [hunk 1]
```diff
@@ -25,7 +25,7 @@
  * crewmember: new onboarding crewmember.
  * rank: new crewmember's rank.
  */
-/datum/team/nation/proc/new_possible_separatist(datum/source, mob/living/crewmember, rank)
+/datum/team/nation/proc/new_possible_separatist(mob/living/crewmember, rank)
```
Comment (Watermelon914) [inline code/modules/antagonists/separatist/separatist.dm:28]: first argument is the SSdcs subsystem, not the crewmeber.

File: code/modules/admin/verbs/secrets.dm [hunk 2]
```diff
@@ -614,3 +606,31 @@
...
+
+/datum/everyone_is_a_traitor_controller/New(objective)
+	src.objective = objective
+	RegisterSignal(SSdcs, COMSIG_GLOB_CREWMEMBER_JOINED, .proc/make_traitor)
+
+/datum/everyone_is_a_traitor_controller/Destroy()
+	UnregisterSignal(SSdcs, COMSIG_GLOB_CREWMEMBER_JOINED)
+	return ..()
+
+/datum/everyone_is_a_traitor_controller/proc/make_traitor(mob/living/player)
```
Comment (Watermelon914) [inline code/modules/admin/verbs/secrets.dm:622]: the first argument is the ssdcs subsystem

**Review by Watermelon914 — CHANGES_REQUESTED** [review]

**Comment by Watermelon914** [comment]
I'm probably gonna ask you to test your code since if you had tested your code you would've noticed that the first argument is ssdcs and not the living player because it would've runtimed.

Please test your code as a contributor, it's vitally important that whatever you are merging is not broken.

**Comment by LemonInTheDark** [comment]
Has this been tested yet?

**Comment by Ghommie** [comment]
I can say I have tested it now.

**Pull request merged**